benchmarks/src/main/java/io/grpc/benchmarks/driver/LoadClient.java

```diff
@@ -0,0 +1,495 @@
+              r = new AsyncUnaryWorker(asyncStubs[i % asyncStubs.length]);
+            } else if (config.getRpcType() == Control.RpcType.STREAMING) {
+              r = new AsyncPingPongWorker(asyncStubs[i % asyncStubs.length]);
+            }
+          }
+          break;
+        }
+        case BYTEBUF_PARAMS: {
+          if (config.getClientType() == Control.ClientType.SYNC_CLIENT) {
+            if (config.getRpcType() == Control.RpcType.UNARY) {
+              r = new GenericBlockingUnaryWorker(channels[i % channels.length]);
+            }
+          } else if (config.getClientType() == Control.ClientType.ASYNC_CLIENT) {
+            if (config.getRpcType() == Control.RpcType.UNARY) {
+              r = new GenericAsyncUnaryWorker(channels[i % channels.length]);
+            } else if (config.getRpcType() == Control.RpcType.STREAMING) {
+              r = new GenericAsyncPingPongWorker(channels[i % channels.length]);
+            }
+          }
+
+          break;
+        }
+        default: {
+          throw new IllegalArgumentException("Not supported");
+        }
+      }
+      if (r == null) {
+        throw new IllegalStateException(config.getRpcType().name()
+            + " not supported for client type "
+            + config.getClientType());
+      }
+      fixedThreadPool.execute(r);
+    }
+  }
+
+  /**
+   * Take a snapshot of the statistics which can be returned to the driver.
+   */
+  Stats.ClientStats getStats() {
+    Histogram intervalHistogram = recorder.getIntervalHistogram();
+
+    Stats.ClientStats.Builder statsBuilder = Stats.ClientStats.newBuilder();
+    Stats.HistogramData.Builder latenciesBuilder = statsBuilder.getLatenciesBuilder();
+    double resolution = 1.0 + Math.max(config.getHistogramParams().getResolution(), 0.01);
+    LogarithmicIterator logIterator = new LogarithmicIterator(intervalHistogram, 1,
+        resolution);
+    double base = 1;
+    while (logIterator.hasNext()) {
+      latenciesBuilder.addBucket((int) logIterator.next().getCountAddedInThisIterationStep());
+      base = base * resolution;
+    }
+    // Driver expects values for all buckets in the range, not just the range of buckets that
+    // have values.
+    while (base < config.getHistogramParams().getMaxPossible()) {
+      latenciesBuilder.addBucket(0);
+      base = base * resolution;
+    }
+    latenciesBuilder.setMaxSeen((double) intervalHistogram.getMaxValue());
+    latenciesBuilder.setMinSeen((double) intervalHistogram.getMinNonZeroValue());
+    latenciesBuilder.setCount(intervalHistogram.getTotalCount());
+    latenciesBuilder.setSum(intervalHistogram.getMean()
+        * intervalHistogram.getTotalCount());
+    // TODO: No support for sum of squares
+
+    statsBuilder.setTimeElapsed(((double)(intervalHistogram.getEndTimeStamp()
+        - intervalHistogram.getStartTimeStamp())) / 1000.0);
+    // Report all the elapsed user-time as system time (which is intentionally incorrect)
```

Comment:
Right, this is all kinds of wrong. Hard to see how we would get system/user time without using something like JVMTI. The only thing I could track down is

OperatingSystemMXBean.getProcessCPUTime

but it doesn't separate user & system time